static_website/main.tf

```diff
@@ -0,0 +1,144 @@
+  acl     = "public-read"
+  policy  = <<EOF
+{
+  "Version": "2008-10-17",
+  "Statement": [
+    {
+      "Sid": "PublicReadForGetBucketObjects",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::cloudfront:user/CloudFront Origin Access Identity ${aws_cloudfront_origin_access_identity.origin_access_identity.id}"
+      },
+      "Action": "s3:GetObject",
+      "Resource": "arn:aws:s3:::${var.origin_bucket}-${var.environment}/*"
+    }
+  ]
+}
+EOF
+
+  website {
+    index_document = "${coalesce(var.index_document, "index.html")}"
+    error_document = "${coalesce(var.error_document, "404.html")}"
+    routing_rules  = <<EOF
+[
+  {
+    "Condition": {
+      "KeyPrefixEquals": "/"
+    },
+    "Redirect": {
+      "ReplaceKeyWith": "${coalesce(var.index_document, "index.html")}"
+    }
+  }
+]
+EOF
+  }
+
+  tags {
+    Name            = "${var.origin_bucket}-origin-bucket"
+    Environment     = "${var.environment}"
+    TechnicalOwner  = "${var.technical_owner}"
+  }
+
+}
+
+resource "aws_s3_bucket" "origin-logs" {
+  bucket  = "${var.origin_bucket}-${var.environment}-logs"
+  acl     = "log-delivery-write"
+
+  tags {
+    Name            = "${var.origin_bucket}-origin-logs"
+    Environment     = "${var.environment}"
+    TechnicalOwner  = "${var.technical_owner}"
+  }
+}
+
+resource "aws_cloudfront_origin_access_identity" "origin_access_identity" {
+    comment = "${var.origin_bucket}-${var.environment}"
+}
+
+resource "aws_cloudfront_distribution" "cdn" {
+  depends_on = [ "aws_s3_bucket.origin"]
+
+  enabled             = true
+  comment             = "Static site for ${var.origin_bucket}-${var.environment}"
+  default_root_object = "${coalesce(var.index_document, "index.html")}"
+  retain_on_delete    = true
+
+  origin {
+    domain_name = "${aws_s3_bucket.origin.bucket}.s3.amazonaws.com"
+    origin_id   = "s3-${aws_s3_bucket.origin.bucket}"
+
+    s3_origin_config {
+      origin_access_identity = "${aws_cloudfront_origin_access_identity.origin_access_identity.cloudfront_access_identity_path}"
+    }
+  }
+
+  aliases = [
+    "${var.origin_bucket}.${var.environment}.${var.region}.${var.account}.${var.nubis_domain}"
+  ]
+
+  logging_config {
+    include_cookies = false
+    prefix          = "${aws_s3_bucket.origin.bucket}"
+    bucket          = "${aws_s3_bucket.origin-logs.bucket_domain_name}"
+  }
+
+  default_cache_behavior {
+    allowed_methods   = ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"]
+    cached_methods    = ["GET", "HEAD"]
+    target_origin_id  = "s3-${aws_s3_bucket.origin.bucket}"
+
+    forwarded_values {
+      query_string = true
+      cookies {
+        forward = "none"
+      }
+    }
+    viewer_protocol_policy = "allow-all"
+    min_ttl                = 0
+    default_ttl            = 30
+    max_ttl                = 86400
+  }
+
+  restrictions {
+    geo_restriction {
+      restriction_type = "none"
+    }
+  }
+
+  viewer_certificate {
```

Comment:
Since most (if not all) static sites are served over SSL, this will need to support non-default SSL certs, unfortunately.